pkg/cloudprovider/aws/instancetypes_test.go

```diff
@@ -0,0 +1,202 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws_test
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	cloudprovideraws "github.com/awslabs/karpenter/pkg/cloudprovider/aws"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/aws/fake"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	instanceTypeMocks = map[string]*ec2.InstanceTypeInfo{
+		"m5.large": {
+			InstanceType:                  aws.String("m5.large"),
+			SupportedUsageClasses:         []*string{aws.String("on-demand")},
+			BurstablePerformanceSupported: aws.Bool(false),
+			BareMetal:                     aws.Bool(false),
+			ProcessorInfo: &ec2.ProcessorInfo{
+				SupportedArchitectures: aws.StringSlice([]string{"x86_64"}),
+			},
+		},
+		"m6g.large": {
+			InstanceType:                  aws.String("m6g.large"),
+			SupportedUsageClasses:         []*string{aws.String("on-demand")},
+			BurstablePerformanceSupported: aws.Bool(false),
+			BareMetal:                     aws.Bool(false),
+			ProcessorInfo: &ec2.ProcessorInfo{
+				SupportedArchitectures: aws.StringSlice([]string{"arm64"}),
+			},
+		},
+	}
+	defaultArch = "amd64"
+	testZone    = "test-zone"
+)
+
+var _ = Describe("InstanceTypes", func() {
+
+	Describe("Getting Instance Types", func() {
+		Context("With amd64 architecture", func() {
+			ec2api, vpcProvider := getInstanceTypeProviderMocks([]string{testZone}, []string{"m5.large"})
+			instanceTypeProvider := cloudprovideraws.NewInstanceTypeProvider(ec2api, &vpcProvider)
+			zonalSubnetOptions := map[string][]*ec2.Subnet{testZone: nil}
+			constraints := &cloudprovider.Constraints{}
+			constraints.Architecture = &v1alpha1.ArchitectureAmd64
+			instanceTypes, err := instanceTypeProvider.Get(context.Background(), zonalSubnetOptions, constraints)
+
+			It("should not error", func() {
```

Comment:
IMO, this should just be checked in the same It() block. You can probably collapse each of the context blocks into single It() blocks

Reply:
👍  yeah it seemed kind of weird to have these in there own `It`, I was just going off what the doc examples were doing, it seemed conventional.  I'll change and shorten up these tests :D